plugin/evm/vm_test.go

```diff
@@ -770,44 +742,60 @@ func TestIssueAtomicTxs(t *testing.T) {
-		t.Fatal(err)
-	}
+	err = blk2.Accept(context.Background())
+	require.NoError(err)
 
-	if status := blk2.Status(); status != choices.Accepted {
-		t.Fatalf("Expected status of accepted block to be %s, but found %s", choices.Accepted, status)
-	}
+	require.Equal(blk2.Status(), choices.Accepted, "Expected status to match")
 
-	if lastAcceptedID, err := vm.LastAccepted(context.Background()); err != nil {
-		t.Fatal(err)
-	} else if lastAcceptedID != blk2.ID() {
-		t.Fatalf("Expected last accepted blockID to be the accepted block: %s, but found %s", blk2.ID(), lastAcceptedID)
-	}
+	lastAcceptedID, err = vm.LastAccepted(context.Background())
+	require.NoError(err)
+	require.Equal(lastAcceptedID, blk2.ID(), "Expected last accepted blockID to be the accepted block")
+
+	vm.blockChain.DrainAcceptorQueue()
 
 	// Check that both atomic transactions were indexed as expected.
 	indexedImportTx, status, height, err := vm.getAtomicTx(importTx.ID())
-	assert.NoError(t, err)
-	assert.Equal(t, Accepted, status)
-	assert.Equal(t, uint64(1), height, "expected height of indexed import tx to be 1")
-	assert.Equal(t, indexedImportTx.ID(), importTx.ID(), "expected ID of indexed import tx to match original txID")
+	require.NoError(err)
+	require.Equal(Accepted, status)
+	require.Equal(uint64(1), height, "expected height of indexed import tx to be 1")
+	require.Equal(indexedImportTx.ID(), importTx.ID(), "expected ID of indexed import tx to match original txID")
 
 	indexedExportTx, status, height, err := vm.getAtomicTx(exportTx.ID())
-	assert.NoError(t, err)
-	assert.Equal(t, Accepted, status)
-	assert.Equal(t, uint64(2), height, "expected height of indexed export tx to be 2")
-	assert.Equal(t, indexedExportTx.ID(), exportTx.ID(), "expected ID of indexed import tx to match original txID")
+	require.NoError(err)
+	require.Equal(Accepted, status)
+	require.Equal(uint64(2), height, "expected height of indexed export tx to be 2")
+	require.Equal(indexedExportTx.ID(), exportTx.ID(), "expected ID of indexed import tx to match original txID")
+
+	// Create processing block and add an atomic tx to it
+	exportTx2, err := vm.newExportTx(vm.ctx.AVAXAssetID, 1, vm.ctx.XChainID, testShortIDAddrs[0], initialBaseFee, []*secp256k1.PrivateKey{testKeys[0]})
+	require.NoError(err)
+
+	err = vm.issueTx(exportTx2, true /*=local*/)
+	require.NoError(err)
+
+	<-issuer
+
+	blk3, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	err = blk3.Verify(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk3.Status(), choices.Processing, "Expected status to match")
+
+	// getAtomicTx should check the processing blocks for any atomic txs.
+	exportTxFetched, status, height, err := vm.getAtomicTx(exportTx2.ID())
+	require.NoError(err)
+	require.Equal(Processing, status)
+	require.Equal(uint64(3), height, "expected height of export tx to be 3")
+	require.Equal(exportTxFetched.ID(), exportTx2.ID(), "expected ID of fetched export tx to match original txID")
```

Comment:
should we further simplify the test by checking the status of `exportTx` before blk2 is accepted instead of creating blk3?